### PR TITLE
Docker for LOH cavatica app

### PR DESCRIPTION
**Pull request by sakshamphul**
## Description

This PR adds dockerfile for LOH (loss of heterozygosity) required for cavatica app. This docker has ubuntu 22.04, python 3.10, bcftools with plugins and some dependent tools  

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Test A: Docker was tested locally, it works fine
- [ ] Test B

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings
- [ ] I have committed any related changes to the PR

### CODE REVIEW

File: LOH/1.0.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM ubuntu:22.04
+LABEL maintainer="Saksham Phul (phuls@chop.edu)"
+
+WORKDIR home/
+
+RUN apt update && apt upgrade -y && \
+        apt install -y wget && \
+        apt install -y python3.10 && \
+        wget https://github.com/samtools/bcftools/releases/download/1.16/bcftools-1.16.tar.bz2 && \
```
Comment (migbro) [inline LOH/1.0.0/Dockerfile:9]: you can just list all the apt packages on the same line, no need to call apt one at a time

File: LOH/1.0.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        apt install bzip2 && \
+        tar -axf bcftools*.tar.bz2 && rm bcftools*.tar.bz2 && bash && \
+        cd bcftools* && apt install -y gcc make liblzma-dev zlib1g-dev libbz2-dev && \
```
Comment (migbro) [inline LOH/1.0.0/Dockerfile:12]: does this cd syntax actually work? `cd bcftools*` is not a well-defined path i don't think

**Review by migbro — CHANGES_REQUESTED** [review]
apt install can be streamlined. `cd` syntax looks funky

**Review by dmiller15 — CHANGES_REQUESTED** [review]
Doesn't build and I'm not sure why. Try breaking up the RUN command and seeing what commands are throwing errors.

File: LOH/1.0.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+LABEL maintainer="Saksham Phul (phuls@chop.edu)"
+
+WORKDIR home/
+
+RUN apt update && apt upgrade -y && \
+        apt install -y wget python3.10 bzip2 pip && \
+        wget https://github.com/samtools/bcftools/releases/download/1.16/bcftools-1.16.tar.bz2 && \
+        tar -axf bcftools*.tar.bz2 && rm bcftools*.tar.bz2 && bash && \
+        cd bcftools-1.16 && \
+	apt install -y gcc make liblzma-dev zlib1g-dev libbz2-dev && \
```
Comment (dmiller15) [inline LOH/1.0.0/Dockerfile:11]: Consolidate these with the other install step.

File: .circleci/config.yml [hunk 4]
```diff
@@ -4338,6 +4368,23 @@ workflows:
...
+            only:
+            - master
+        name: LOH_1.0.0_merge
+        context: dockerhub-vars
+        deploy: true
+        registry: pgc-images.sbgenomics.com
+        image: d3b-bixu/loh
+        tag: 1.0.0
+        path: LOH/1.0.0/
+        docker-context: LOH/1.0.0/
```
Comment (dmiller15) [inline .circleci/config.yml:4383]: add the extra build commands here as well.

**Review by dmiller15 — CHANGES_REQUESTED** [review]

**Review by migbro — APPROVED** [review]
Looks much improved, builds with circleci

**Review by migbro — COMMENTED** [review]
Rolling back approval since Dan pointed out to me that circleci is struggling to build an ubuntu 22 image. I leave it up to you an @dmiller15 how much you want to make this work (will likely need to some time in the future). Otherwise, could build with ubuntu 20.04 perhaps? That is still supported

**Review by dmiller15 — APPROVED** [review]

**Comment by sakshamphul** [comment]
> Agreed on the rolling back to 20.04. It's not particularly close to being phased out and by the time that happens maybe this issue will be resolved. I'll continue to look into the issue and make a PR if I find a solution.

Ubuntu 20:04 passed all the 125 test. @Take away: Circleci does not work with Ubuntu 22:04

**Pull request merged**